### PR TITLE
Add `@builder` decorator as alternative to `build()` classmethod

### PR DESCRIPTION
**Pull request by maackle**

### CODE REVIEW

**Comment by maackle** [comment]
The linter is failing in CI but runs fine locally, not sure why.

File: apistar/commands/schema.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def schema(format: Format) -> None:  # pragma: nocover
     from apistar.cli import get_current_app
     from apistar.apischema import APISchema
     app = get_current_app()
-    schema = APISchema.build(app)
+    schema = core.get_builder(APISchema)(app)
```
Comment (tomchristie) [inline apistar/commands/schema.py]: Just use `build_apischema` directly here, for now.

File: apistar/settings.py [hunk 2]
```diff
@@ -21,11 +18,14 @@ def get(self, indexes: Union[str, List[str]], default: Any=None) -> Any:
...
-        return args[0]
+@builder
+def build_settings(app: app.App) -> Settings:
+    return Settings(app.settings)
 
-    @classmethod
-    def build(cls, arg_name: ArgName, settings: Settings):
-        return settings.get(arg_name)
+
+Setting = object
```
Comment (tomchristie) [inline apistar/settings.py]: Does `Setting = typing.Any` work?

File: tests/test_apischema.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def set_complete(ident: int, complete: bool):  # pragma: nocover
 
 
 def test_api_schema():
-    schema = APISchema.build(app)
+    schema = core.get_builder(APISchema)(app)
```
Comment (tomchristie) [inline tests/test_apischema.py]: Just build_apischema directly would be good.

File: apistar/settings.py (hunk 2 above)
Comment (maackle) [inline apistar/settings.py]: Nope, because later on this gets passed through `issubclass`, which expects a Class, not a type.

**Comment by tomchristie** [comment]
Obsoleted by #235 - We won't have `build` methods at all once this work comes in.

**Pull request closed without merging**